src/garage/torch/algos/maml.py

```diff
@@ -0,0 +1,383 @@
+"""Model-Agnostic Meta-Learning (MAML) algorithm implementation for RL."""
+import collections
+
+from dowel import tabular
+import numpy as np
+import torch
+
+from garage.misc import tensor_utils
+from garage.sampler import OnPolicyVectorizedSampler
+from garage.tf.samplers import BatchSampler
+from garage.torch.algos import _Default, make_optimizer, VPG
+from garage.torch.optimizers import DiffSGD
+from garage.torch.utils import update_module_params
+
+
+class MAML:
+    """Model-Agnostic Meta-Learning (MAML).
+
+    Args:
+        env (garage.envs.base): A gym environment.
+        policy (garage.torch.policies.base.Policy): Policy.
+        baseline (garage.np.baselines.Baseline): The baseline.
+        meta_optimizer (Union[type, tuple[type, dict]]): Type of optimizer.
+            This can be an optimizer type such as `torch.optim.Adam` or a
+            tuple of type and dictionary, where dictionary contains arguments
+            to initialize the optimizer e.g. `(torch.optim.Adam, {'lr' = 1e-3})`  # noqa: E501
+        meta_batch_size (int): Number of tasks.
+        inner_lr (double): Adaptation learning rate.
+        num_grad_updates (int): Number of adaptation gradient steps.
+        inner_algo (garage.torch.algos.vpg): The inner algorithm used for
+            computing loss.
+
+    """
+
+    def __init__(self,
+                 env,
+                 policy,
+                 baseline,
+                 meta_optimizer=torch.optim.Adam,
+                 meta_batch_size=40,
+                 inner_lr=0.1,
+                 num_grad_updates=1,
+                 inner_algo=None):
+        if inner_algo is None:
```

Comment:
this is a dangerous default to me. most people don't want to run MAML with VPG. just make this a required argument.

Reply:
I think there is a misunderstanding here. The `inner_algo` is used only to compute the inner loss (inner KL is also computed, but only for logging purpose), so VPG and PPO should be the only two valid algorithms. The variant usually goes into meta optimization (e.g. the TRPO in ProMP-TRPO refers to the meta optimization).